tests/ttnn/unit_tests/operations/test_matmul.py

```diff
@@ -1170,3 +1171,176 @@ def test_sd_matmul(device, batch_size, channel_a, channel_b, m_size, k_size, n_s
+        x = min(num_cores, 8)
+        y = max(1, num_cores // 8)
+        core_grid = (x, y)
+        return core_grid
+
+    def core_range_for_num_cores(num_cores):
+        core_grid = core_grid_size_for_num_cores(num_cores)
+        core_range = ttnn.CoreRangeSet(
+            {
+                ttnn.CoreRange(
+                    ttnn.CoreCoord(0, 0),
+                    ttnn.CoreCoord(core_grid[0] - 1, core_grid[1] - 1),
+                ),
+            }
+        )
+        return core_range
+
+    TILE_SIZE = 32
+
+    if config == "tg_llama_FF1":
+        assert M is None and K is None and N is None, "Cannot specify config and any of M, K, N"
+        cluster_size = (4, 8)
+        hidden_size = 8192
+        ff_size = 28 * 1024
+        M = 32
+        K = padded_size_per_device_for_num_cores(hidden_size, cluster_size[0], num_activation_cores)
+        N = ff_size // cluster_size[1]
+        N_padded = padded_size_per_device_for_num_cores(ff_size, cluster_size[1], num_compute_cores)
+    else:
+        assert M is not None and K is not None and N is not None, "Must specify M, K, N"
+        N_padded = N
+
+    # Weights
+    mem_config_weights = ttnn.MemoryConfig(
+        ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+        ttnn.BufferType.L1,
+        ttnn.ShardSpec(
+            core_range_for_num_cores(num_compute_cores),
+            [
+                K,
+                N_padded // num_compute_cores,
+            ],
+            ttnn.ShardOrientation.ROW_MAJOR,
+            False,
+        ),
+    )
+
+    if has_bias:
+        mem_config_bias = ttnn.MemoryConfig(
+            ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+            ttnn.BufferType.L1,
+            ttnn.ShardSpec(
+                core_range_for_num_cores(num_compute_cores),
+                [
+                    32,
+                    N_padded // num_compute_cores,
+                ],
+                ttnn.ShardOrientation.ROW_MAJOR,
+                False,
+            ),
+        )
+
+    # Input
+    mem_config_input = ttnn.MemoryConfig(
+        ttnn.TensorMemoryLayout.WIDTH_SHARDED,
+        ttnn.BufferType.L1,
+        ttnn.ShardSpec(
+            core_range_for_num_cores(num_activation_cores),
+            [
+                M,
+                K // num_activation_cores,
+            ],
+            ttnn.ShardOrientation.ROW_MAJOR,
+            False,
+        ),
+    )
+
+    compute_kernel_config = ttnn.WormholeComputeKernelConfig(
+        math_fidelity=ttnn.MathFidelity.LoFi,
+        math_approx_mode=True,
+        fp32_dest_acc_en=True,
+        packer_l1_acc=True,
+    )
+
+    input_shape = [1, 1, M, K]
+    input_tensor = torch.randn(input_shape).float()
+    tt_input_tensor = ttnn.as_tensor(
+        input_tensor,
+        dtype=in0_dtype,
+        layout=ttnn.TILE_LAYOUT,
+        device=device,
+        memory_config=mem_config_input,
+    )
+
+    weights_tensor = torch.randn([1, 1, K, N]).float()
```

Comment:
1. I think you can get rid of `.float()`.. should return `torch.float32` by default
2. You probably want to set torch seed in this test file as well
3. You probably want to use a test-fixture to specify wormhole config or move this unit test somewhere so regressions don't complain

Reply:
3. Do you mean to only run it for wormhole on CI? Shouldn't `@run_for_wormhole_b0()` do that?